src/sdk/pynni/nni/analysis_utils/sensitivity/torch/sensitivity_analysis.py

```diff
@@ -0,0 +1,257 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import os
+import copy
+import csv
+import logging
+from collections import OrderedDict
+import matplotlib
+import matplotlib.pyplot as plt
+from matplotlib.lines import Line2D
+
+import numpy as np
+import torch.nn as nn
+
+from nni.compression.torch import LevelPruner
+from nni.compression.torch import L1FilterPruner
+from nni.compression.torch import L2FilterPruner
+
+# use Agg backend
+matplotlib.use('Agg')
+SUPPORTED_OP_NAME = ['Conv2d', 'Conv1d']
+SUPPORTED_OP_TYPE = [getattr(nn, name) for name in SUPPORTED_OP_NAME]
+
+logger = logging.getLogger('Sensitivity_Analysis')
+logger.setLevel(logging.INFO)
+
+
+class SensitivityAnalysis:
+    def __init__(self, model, val_func, sparsities=None, prune_type='l1', early_stop=1.0):
+        """
+        Perform sensitivity analysis for this model.
+        Parameters
+        ----------
+            model:
+                the model to perform sensitivity analysis
+            val_func:
+                validation function for the model. Due to
+                different models may need different dataset/criterion
+                , therefore the user need to cover this part by themselves.
+                val_func take the model as the first input parameter, and
+                return the accuracy as output.
+            sparsities:
+                The sparsity list provided by users.
```

Comment:
what is the meaning of this list? each element is a layer's sparsity?

Reply:
what is the meaning of the three numbers?

Reply:
They are sparsity values which means how much weight the pruner should prune.  Take [0.25, 0.5, 0.75] for an example, the SensitivityAnalysis will prune 25% 50% 75% weights gradually for each layer(SensitivityAnalysis only prune a layer once a time, the other layers are set to their original weights), and record the model's accuracy at the same time.

Reply:
got it, thanks. please make it clear in docstring and doc.

Reply:
Ok~ :)